compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -336,76 +307,78 @@ public CodeAnalyzer(CompilerContext context) {
         this.names = Names.getInstance(context);
         this.symResolver = SymbolResolver.getInstance(context);
         this.reachabilityAnalyzer = ReachabilityAnalyzer.getInstance(context);
-        this.enableExperimentalFeatures = Boolean.parseBoolean(
```

Comment:
[EXPERIMENTAL]
Shall we keep this and related code in CodeAnalyzer we can create an issue to track these. If there is a experimental feature in future we may have to rewrite the code again. 
Also proper removal needs to delete this from compiler options as well.
Another way is to move the logic to a util class and share between all the visitors but atm only used in CodeAnalyzer

Reply:
Referring the issue https://github.com/ballerina-platform/ballerina-lang/issues/34776

Reply:
https://github.com/ballerina-platform/ballerina-lang/issues/34776